### PR TITLE
Add missing serviceName in fleet-agent statefulset

### PR DESCRIPTION
**Pull request by p-se**
Refers to #2048

### CODE REVIEW

File: internal/cmd/controller/agentmanagement/agent/manifest.go [hunk 1]
```diff
@@ -115,9 +115,23 @@ func Manifest(namespace string, agentScope string, opts ManifestOptions) []runti
 		},
 	}
```
Comment (manno) [inline internal/cmd/controller/agentmanagement/agent/manifest.go:117]:
```suggestion

// added for compatibility with old helm versions, the service is not used and not necessary
```

**Comment by manno** [comment]
Discussed, we don't need to support (now) very old versions of kubectl.

**Comment by manno** [comment]
This adds a headless service to the fleet-agent statefulset.

    StatefulSets currently require a Headless Service to be responsible for the network identity of the Pods. You are responsible for creating this Service.

https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/

If the service is missing, Rancher shows a warning in the UI.

File: internal/cmd/controller/agentmanagement/agent/manifest.go [hunk 2]
```diff
@@ -115,9 +115,25 @@ func Manifest(namespace string, agentScope string, opts ManifestOptions) []runti
...
 
+	// Added for compatibility with old helm versions, the service is not used
+	// and not necessary.
```
Comment (manno) [inline internal/cmd/controller/agentmanagement/agent/manifest.go:119]:
```suggestion
	// Statefulsets require a headless service. The service is not used.
```

File: e2e/single-cluster/sharding_test.go [hunk 3]
```diff
@@ -100,36 +100,38 @@ var _ = Describe("Filtering events by shard", Label("infra-setup"), Ordered, fun
...
-					} else {
-						Expect(hasReconciledGitRepo).To(BeFalseBecause(
-							"GitRepo %q labeled with shard %q should not have been"+
-								" deployed by controller for shard %q",
-							gitrepoName,
-							shard,
-							s,
-						))
-					}
+					Eventually(func(g Gomega) {
```
Comment (weyfonk) [inline e2e/single-cluster/sharding_test.go:103]: Nice one, thanks for looking into this :muscle:

Comment (p-se) [reply e2e/single-cluster/sharding_test.go:103]: Glad you like it :slightly_smiling_face:

**Review by manno — APPROVED** [review]

**Pull request merged**